packages/dds/tree/src/simple-tree/tree.ts

```diff
@@ -84,10 +88,47 @@ export interface ITreeConfigurationOptions {
 	 * bit slower.
 	 */
 	enableSchemaValidation?: boolean;
+
+	/**
+	 * If `true`, constructing the configuration will error if any valid input data could parse ambiguously
+	 * (type of the node could fail to be inferred from content, thus requiring use of {@link Unhydrated} nodes to disambiguate).
+	 *
+	 * @defaultValue `false`.
+	 *
+	 * @remarks
+	 * When this is true, it ensures that the compile time type safety for data when constructing nodes is sufficient to ensure than the runtime behavior will not give node data ambiguity errors.
+	 *
+	 * This ensures that the canonical JSON like representation of the node of any union in the tree is lossless and unambiguous.
+	 * This canonical JSON like representation consists of arrays, plain old JavaScript objects with string keys, booleans, numbers (excluding NaN, -0 and infinities), strings, null and IFluidHandles.
+	 * Assuming any IFluidHandles get special handling, its JSON compatible, as well as compatible with the node creation APIs (such as schema class constructors).
+	 *
+	 * Currently these cases can cause ambiguity in a union:
+	 *
+	 * - More than one ArrayNode type: it's impossible to tell which array type is intended in the case of empty arrays (`[]`).
```

Comment:
Each of these cases would be great to have as its own `@example`. I can help write them if you like :)

Reply:
I added one set of examples. Maybe once we are happy with the other aspect of these docs we can flesh it out with a few more.